### PR TITLE
attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves

### PR DESCRIPTION
**Pull request by Zulusssss**
Add law and tests.

### CODE REVIEW

File: symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+## by an insulating material or an air gap. It is used to transmit radio frequency electrical signals.
+## The specific resistance of a coaxial waveguide depends on the radius of the outer conductor and the radius of the inner conductor,
+## as well as on the relative permeability of the insulator material, frequency of signal and specific conductivity of conductor.
+## The attenuation coefficient shows how many times the transmitted signal weakens per unit length of the coaxial waveguide.
+
+## Law is: am = 2 * Rs * (n^2 * (b / a)^3 + m^2) / (Z0 * b * sqrt((1 - (L / L1)^2) * (n^2 * (b / a)^2 + m^2))), where
+## am - attenuation coefficient in metal,
+## Rs - surface resistance,
+## m - first index,
+## n - second index,
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py:17]: Index of what?
Add description for m and n indexes.

File: symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+## a - height of the waveguide cross section,
+## b - width of the waveguide cross section,
+## Z0 - characteristic resistance of the material filling the waveguide,
+## L - wavelength,
+## L1 - critical wavelength.
+
+
+attenuation_coefficient = Symbol("attenuation_coefficient", dimensionless / units.length)
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py:25]:
```suggestion
attenuation_coefficient = Symbol("attenuation_coefficient", 1 / units.length)
```

File: symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+# Description
+## A coaxial waveguide is an electrical cable consisting of a central conductor and a shield arranged coaxially and separated
+## by an insulating material or an air gap. It is used to transmit radio frequency electrical signals.
+## The specific resistance of a coaxial waveguide depends on the radius of the outer conductor and the radius of the inner conductor,
+## as well as on the relative permeability of the insulator material, frequency of signal and specific conductivity of conductor.
+## The attenuation coefficient shows how many times the transmitted signal weakens per unit length of the coaxial waveguide.
+
+## Law is: am = 2 * Rs * (n^2 * (b / a)^3 + m^2) / (Z0 * b * sqrt((1 - (L / L1)^2) * (n^2 * (b / a)^2 + m^2))), where
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py:13]: Show the link to the law and test data if possible. There are many ways to make a mistake in this formula so I want to double check it.

Comment (Zulusssss) [reply symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py:13]: > Show the link to the law and test data if possible. There are many ways to make a mistake in this formula so I want to double check it.

This is a reference to the law. Page 8.
https://core.ac.uk/download/pdf/49213156.pdf

But I can't find a test case.

Comment (blackyblack) [reply symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py:13]: I do not see anything related on page 8. Screenshot maybe?

Comment (Zulusssss) [reply symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py:13]: > I do not see anything related on page 8. Screenshot maybe?

I sent the wrong link. Page 8.
http://old.istu.ru/files/material-static/1664/glava07.pdf

image

Comment (blackyblack) [reply symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py:13]: Well, it does not correspond to the code you wrote.

Comment (Zulusssss) [reply symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py:13]: > Well, it does not correspond to the code you wrote.

Yes, I have already noticed and corrected it.

Comment (blackyblack) [reply symplyphysics/laws/electricity/circuits/attenuation_coefficient_in_metal_in_rectangular_waveguide_for_transverse_magnetic_waves.py]: Still does not correspond to description.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**